diff --git a/tools/CMakeLists.txt b/tools/CMakeLists.txt
index 0b710050..e1a5b3d0 100644
--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -8,6 +8,7 @@ include_directories(${CMAKE_CURRENT_BINARY_DIR})
 protobuf_generate_cpp(CAFFE_PROTO_SRCS CAFFE_PROTO_HDRS caffe.proto)

 add_executable(caffe2ncnn caffe2ncnn.cpp ${CAFFE_PROTO_SRCS} ${CAFFE_PROTO_HDRS})
+include_directories(${Protobuf_INCLUDE_DIR})

 target_link_libraries(caffe2ncnn ${PROTOBUF_LIBRARIES})

diff --git a/tools/caffe2ncnn.cpp b/tools/caffe2ncnn.cpp
index 0eff756a..229cc653 100644
--- a/tools/caffe2ncnn.cpp
+++ b/tools/caffe2ncnn.cpp
@@ -193,7 +193,7 @@ static bool read_proto_from_binary(const char* filepath, google::protobuf::Messa
     google::protobuf::io::IstreamInputStream input(&fs);
     google::protobuf::io::CodedInputStream codedstr(&input);

-    codedstr.SetTotalBytesLimit(INT_MAX, INT_MAX / 2);
+    codedstr.SetTotalBytesLimit(INT_MAX);

     bool success = message->ParseFromCodedStream(&codedstr);
